### PR TITLE
implement multiple belga coverages in single item

### PR DESCRIPTION
**Pull request by petrjasek**
SDBELGA-519 SDBELGA-452

### CODE REVIEW

File: client/extensions/belgaCoverage/src/belga-coverage-preview.tsx [hunk 1]
```diff
@@ -2,28 +2,28 @@ import * as React from 'react';
...
-                        {props.item.type !== 'graphic' &&
-                            <BelgaCoverageInfo coverageId={id} superdesk={superdesk} />
-                        }
-                        <BelgaCoverageImages
-                            coverageId={id}
-                            rendition={'preview'}
-                            maxImages={10}
-                            superdesk={superdesk}
-                        />
+                        {props.item.type !== 'graphic' && (
```
Comment (tomaskikutis) [inline client/extensions/belgaCoverage/src/belga-coverage-preview.tsx:16]: Is it really supposed to work for all types except graphic?

If so, it'd be nicer to use a ternary

```
{
    props.item.type === 'graphic'
        ? (
            <Component1 />
        )
        : (
            <Component2 />
        )
}
```

File: client/extensions/belgaCoverage/src/belga-coverage-carousel.tsx [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    }
+
+    render() {
+        if (this.state.loading) {
+            return (
+                <div className="sd-loader" />
+            );
+        }
+
+        const {Alert} = this.props.superdesk.components;
```
Comment (tomaskikutis) [inline client/extensions/belgaCoverage/src/belga-coverage-carousel.tsx:57]: If you are targeting `develop` branch of the client, it's now possible to import `superdesk` in [the root scope of the file](https://github.com/tomaskikutis/superdesk-planning/blob/assignments-widget-in-the-toolbar/client/planning-extension/src/assignments-overview/assignments-overview-list-item.tsx#L7) now

File: client/extensions/belgaCoverage/src/belga-coverage-carousel.tsx [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+                <div className="sd-thumb-carousel__item-inner">
+                    <img src={props.thumbnailUrl} alt={props.name} />
+                </div>
+            </div>
+        );
+
+        const headerTemplate = (
+            <div className="sd-thumb-carousel__header">
+                <h4 className="sd-thumb-carousel__heading">{coverage.name}</h4>
+                <Badge text={'' + coverage.nrImages} type='light' />
```
Comment (tomaskikutis) [inline client/extensions/belgaCoverage/src/belga-coverage-carousel.tsx:81]: Is `nrImages` image count? It'd be good to rename it or at least add a comment in the interface if the name comes from their API.

File: client/extensions/belgaCoverage/src/belga-coverage-preview.tsx [hunk 4]
```diff
@@ -2,28 +2,27 @@ import * as React from 'react';
...
-                        }
-                        <BelgaCoverageImages
-                            coverageId={id}
-                            rendition={'preview'}
-                            maxImages={10}
-                            superdesk={superdesk}
-                        />
+                        {props.item.type !== 'graphic' ? ( // belga coverage in an article
+                            <BelgaCoverageCarousel coverageId={id} superdesk={superdesk} preview={true} />
+                        ) : ( // single belga coverage in the search, there is description already
```
Comment (tomaskikutis) [inline client/extensions/belgaCoverage/src/belga-coverage-preview.tsx:18]: What kind of description are you referring to here?

> belga coverage in the search

do you mean preview pane opened when in search view?

To me it seems like both components render lists of images, why are different components used?

**Review by tomaskikutis — APPROVED** [review]

File: client/extensions/belgaCoverage/src/belga-coverage-preview.tsx (hunk 4 above)
Comment (petrjasek) [reply client/extensions/belgaCoverage/src/belga-coverage-preview.tsx:18]: we have search for those coverages, in that case in the preview you open just single coverage item which has name as headline and description as abstract and we only need to show images in preview on top of it, while in the other case it's embedded in an article which doesn't have the coverage name/description in the other metadata, so we display it all together

**Pull request merged**